frontend/src/metabase/lib/expressions/typechecker.js

```diff
@@ -28,6 +28,34 @@ export function typeCheck(cst, startRule) {
       return result;
     }
 
+    logicalOrExpression(ctx) {
+      const type = ctx.operands.length > 1 ? "boolean" : this.typeStack[0];
+      this.typeStack.unshift(type);
+      const result = super.logicalOrExpression(ctx);
+      this.typeStack.shift();
+      return result;
+    }
+    logicalAndExpression(ctx) {
+      const type = ctx.operands.length > 1 ? "boolean" : this.typeStack[0];
```

Comment:
What's this for? Can you have unary AND/OR in some situations?

Reply:
This is another Chevrotain trait, a syntax node with only a single operand is essentially just a pass-through. Thus, we only propagate the type only for `A AND B` construct.